public/js/components/Expressions.js

```diff
@@ -0,0 +1,69 @@
+const React = require("react");
+const { connect } = require("react-redux");
+const { bindActionCreators } = require("redux");
+const ImPropTypes = require("react-immutable-proptypes");
+const classnames = require("classnames");
+const actions = require("../actions");
+const { getExpressions, getPause } = require("../selectors");
+// const { truncateStr } = require("../utils/utils");
+const { DOM: dom, PropTypes } = React;
+
+require("./Expressions.css");
+
+const Expressions = React.createClass({
+  propTypes: {
+    pauseInfo: ImPropTypes.map,
+    expressions: ImPropTypes.list,
+    addExpression: PropTypes.func,
+    evaluateExpression: PropTypes.func,
+    command: PropTypes.func
+  },
+
+  displayName: "Expressions",
+
+  componentWillUpdate(nextProps, nextState) {
+    const pauseInfo = nextProps.pauseInfo;
+    const expressions = nextProps.expressions.toJS();
+    if (pauseInfo) {
+      for (let expression of expressions) {
+        nextProps.evaluateExpression(expression);
+      }
```

Comment:
I don't think we want to re-evaluate all of the expressions here. It is probably better to re-evaluate expressions in the pause action.